gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/impl/sql/GenericStatement.java

```diff
@@ -833,6 +832,10 @@ private final PreparedStatement prepMinion(
                                             qinfo = handleInsertAndInsertSubSelect(qinfo, qt);
                                           }
 
+                                          if (qinfo != null && qinfo.isDML() && invalidQueryOnColumnTable(lcc, (DMLQueryInfo)qinfo)) {
```

Comment:
Same as above. Ypu want to put a isSnappyStore check. For row store mode call to this function 'invalidQueryOnColumnTable' will all be useless. Right?

Reply:
Ok